libs/src/sdk/sdk.ts

```diff
@@ -153,18 +153,22 @@ const initializeApis = ({
 }) => {
   const initializationPromise = discoveryProvider.init()
 
-  const fetchApi = async (url: string) => {
+  const fetchApi = async (url: string, context?: RequestInit) => {
     // Ensure discovery node is initialized
     await initializationPromise
 
     // Append the appName to the query params
     const urlWithAppName =
       url + (url.includes('?') ? '&' : '?') + querystring({ app_name: appName })
 
+    const requestParams: Record<string, unknown> = {}
+    if (!!context) {
+      Object.assign(requestParams, context)
```

Comment:
the rare literal `Object.assign` 😅 

I was thinking more like `requestParams = { ...context, endpoint: urlWithAppName }`?

Doesn't really matter though, can ignore